api/handlers.go

```diff
@@ -2454,22 +2454,17 @@ func prepareNestedSources(cm cache.CacheManager, exec boil.Executor, f SourcesFi
 			}
 		}
 
-		var uids pq.StringArray
-		q := `SELECT array_agg(DISTINCT s.uid)
-              FROM authors a INNER JOIN authors_sources "as" ON a.id = "as".author_id
-              INNER JOIN sources s ON "as".source_id = s.id
-              WHERE a.code = ANY($1)`
-		err := queries.Raw(exec, q, pq.Array(f.Authors)).QueryRow().Scan(&uids)
-		if err != nil {
-			return nil, nil, err
-		}
-		sourceUids = append(sourceUids, uids...)
+		roots := cm.AuthorsStats().GetSources(f.Authors)
+		sourceUids = append(sourceUids, roots...)
 	}
 
 	// blend in requested sources
 	sourceUids = append(sourceUids, f.Sources...)
 
-	uids, ids := cm.SourcesStats().GetChildren(sourceUids)
+	uids, ids := cm.SourcesStats().GetTree().GetChildren(sourceUids)
+
+	uids = utils.ClearDuplicateString(uids)
```

Comment:
Could be nice to move the duplication removals into tree and have a variant method `GetUniqueChildren`.
btw, we don't need the uniqueness on tags ? why only in sources filter ?

Reply:
add and use GetUniqueChildren